src/main/java/com/ecommerce/ECommerceApp/model/ShippingDetails.java

```diff
@@ -46,8 +46,12 @@ public LocalDateTime getDeliveryDate() {
      * @param deliveryDate the delivery date
      */
     public void setDeliveryDate(LocalDateTime deliveryDate) {
+        if (deliveryDate.isBefore(LocalDateTime.now())) {
+            throw new IllegalArgumentException("Delivery date cannot be in the past.");
+        }
```

Comment:
Added validation for the delivery date to ensure dates cannot be set in the past, improving data integrity. This change aligns with best practices outlined in ["Java Clean Code"](https://www.baeldung.com/java-clean-code) by Baeldung, which emphasizes the importance of input validation for reliability and user experience.